presto-main/src/main/java/com/facebook/presto/operator/scalar/StringFunctions.java

```diff
@@ -617,6 +617,45 @@ public static Slice rightPad(@SqlType("varchar(x)") Slice text, @SqlType(Standar
         return pad(text, targetLength, padString, text.length());
     }
 
+    @Description("computes Levenshtein distance between two strings")
+    @ScalarFunction
+    @SqlType(StandardTypes.BIGINT)
+    public static long levenshteinDistance(@SqlType(CodePointsType.NAME) int[] leftCodePoints, @SqlType(CodePointsType.NAME) int[] rightCodePoints)
+    {
+        if (leftCodePoints.length < rightCodePoints.length) {
+            int[] tempCodePoints = leftCodePoints;
+            leftCodePoints = rightCodePoints;
+            rightCodePoints = tempCodePoints;
+        }
+
+        if (rightCodePoints.length == 0) {
+            return leftCodePoints.length;
+        }
+
+        int[] distances = new int[rightCodePoints.length + 1];
+        for (int i = 0; i < rightCodePoints.length + 1; i++) {
+            distances[i] = i;
+        }
+
+        for (int i = 0; i < leftCodePoints.length; i++) {
+            int leftDist = i + 1;
```

Comment:
It looks like `leftDist` is always `distnace[j]` in the inner loop (except after the last line). As a result, this variable appears unnecessary.

It also looks like you didn't read from or write to `distances[0]` at all. Is this intentional?